### PR TITLE
Encrypt state

### PR DESCRIPTION
**Pull request by rupertbates**
## Why are you doing this?

For data protection reasons, we want to encrypt the state we pass in to step functions. This PR does that.

[**Trello Card**](https://trello.com/c/c6wlfctI/631-add-step-function-encryption-support-to-support-frontend)

### CODE REVIEW

File: app/lib/stepfunctions/StateMachineContainer.scala [hunk 1]
```diff
@@ -18,7 +19,7 @@ object StateMachineContainer {
       case _: StateMachineDeletingException => RetryWithNewMachine.asLeft
       case _: ExecutionLimitExceededException => Fail.asLeft
       case _: ExecutionAlreadyExistsException => Fail.asLeft
-      case _: InvalidExecutionInputException => Fail.asLeft
+      case e: InvalidExecutionInputException => Fail.asLeft
```
Comment (davidfurey) [inline app/lib/stepfunctions/StateMachineContainer.scala]: Presumably changed for debugging, could go back to `_`

File: cloud-formation/cfn.yaml [hunk 2]
```diff
@@ -124,6 +124,14 @@ Resources:
...
                 - states:StartExecution
                 Resource: arn:aws:states:*:*:*
+        - PolicyName: KMSEncryption
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                - kms:Encrypt
+                Resource: arn:aws:kms:eu-west-1:865473395570:key/89f17663-c79e-4104-b718-ee0986402ad9
```
Comment (davidfurey) [inline cloud-formation/cfn.yaml]: We should probably use a different key for CODE and PROD

File: app/lib/stepfunctions/StateMachineContainer.scala (hunk 1 above)
Comment (rupertbates) [inline app/lib/stepfunctions/StateMachineContainer.scala]: 👍

File: cloud-formation/cfn.yaml (hunk 2 above)
Comment (rupertbates) [inline cloud-formation/cfn.yaml]: We do and I need to add it here, well spotted :1st_place_medal:

File: cloud-formation/cfn.yaml [hunk 3]
```diff
@@ -124,6 +124,16 @@ Resources:
...
                 Resource: arn:aws:states:*:*:*
+        - PolicyName: KMSEncryption
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                - kms:Encrypt
+                Resource:
+                - arn:aws:kms:eu-west-1:865473395570:key/89f17663-c79e-4104-b718-ee0986402ad9
```
Comment (davidfurey) [inline cloud-formation/cfn.yaml]: If you stored this in a stage variable, CODE would only be able to use CODE kms and PROD would only be able to use PROD kms

Comment (rupertbates) [inline cloud-formation/cfn.yaml]: Yup good idea

**Pull request merged**